src/services/offer.js

```diff
@@ -64,6 +64,13 @@ class OfferService {
         ).map((o) => o.withCompany()));
 
     }
+
+    async getOfferByID({ id = 0 }) {
+        const offer = await Offer.findById(id).current();
```

Comment:
`.current()` should be removed since you are fetching an entity by Id, therefore you don't want to filter by date, sometimes you might want to just fetch it and do some operations, regardless if it is expired or not (maybe we'll want to "revive" offers in the future for example)